application/src/main/java/org/togetherjava/tjbot/commands/filesharing/FileSharingMessageListener.java

```diff
@@ -0,0 +1,183 @@
+        String fileName = attachment.getFileName();
+        String fileExtension = attachment.getFileExtension();
+
+        if (fileExtension == null || fileExtension.equals("txt")) {
+            fileExtension = "java";
+        } else if (fileExtension.equals("fxml")) {
+            fileExtension = "xml";
+        }
+
+        fileName = fileName.substring(0, fileName.lastIndexOf(".")) + "." + fileExtension;
+        return fileName;
+    }
+
+    private @NotNull String uploadToGist(@NotNull GistRequest jsonRequest) {
+        String body;
+        try {
+            body = JSON.writeValueAsString(jsonRequest);
+        } catch (JsonProcessingException e) {
+            throw new IllegalStateException("Couldn't parse json request!", e);
+        }
+
+        HttpRequest request = HttpRequest.newBuilder()
+            .uri(URI.create(SHARE_API))
+            .header("Accept", "application/json")
+            .header("Authorization", "token " + gistApiKey)
+            .POST(HttpRequest.BodyPublishers.ofString(body))
+            .build();
+
+        HttpResponse<String> apiResponse;
+        try {
+            apiResponse = CLIENT.send(request, HttpResponse.BodyHandlers.ofString());
+        } catch (IOException e) {
+            throw new UncheckedIOException(e);
+        } catch (InterruptedException e) {
+            Thread.currentThread().interrupt();
+            throw new IllegalStateException("HttpResponse interrupted!", e);
+        }
+
+        int statusCode = apiResponse.statusCode();
+
+        if (statusCode < HttpURLConnection.HTTP_OK
+                || statusCode >= HttpURLConnection.HTTP_MULT_CHOICE) {
+            throw new IllegalStateException("Gist API unexpected response: " + apiResponse.body());
+        }
+
+        GistResponse gistResponse;
+        try {
+            gistResponse = JSON.readValue(apiResponse.body(), GistResponse.class);
+        } catch (JsonProcessingException e) {
+            throw new IllegalStateException("Couldn't parse response!", e);
+        }
+        return gistResponse.getHtmlUrl();
+    }
+
+    private void sendResponse(@NotNull MessageReceivedEvent event, @NotNull String url) {
+        Message message = event.getMessage();
+        String replyContent =
+                "I uploaded your file(s) as gist, it is much easier to read for everyone like that, especially for mobile users";
+
+        message.reply(replyContent).setActionRow(Button.link(url, "gist")).queue();
+    }
+
+    private boolean isHelpThread(@NotNull MessageReceivedEvent event) {
+        if (event.getChannelType() != ChannelType.GUILD_PUBLIC_THREAD) {
+            return false;
+        }
+
+        ThreadChannel thread = event.getThreadChannel();
+        String rootChannelName = thread.getParentChannel().getName();
+        return isStagingChannelName.test(rootChannelName)
+                || isOverviewChannelName.test(rootChannelName);
+    }
```

Comment:
Unsure, but isnt such a method already available in `HelpSystemHelper` (or whatever its called)? If so, please use the helper instead.

Reply:
couldn't find it, but I stole it somewhere but I think it was a non-static method thats why I simply copied it

Reply:
> Unsure, but isnt such a method already available in HelpSystemHelper (or whatever its called)? If so, please use the helper instead.

its in a diff package and the method is not public, also the use case r different the method in HelpSystemHelper sends a message if its not a help thread + it is getting removed in my pr so its gud if they make their own method

Reply:
fair